ta/pkcs11/src/processing_ec.c

```diff
@@ -646,3 +646,102 @@ size_t ecdsa_get_input_max_byte_size(TEE_OperationHandle op)
 		return 0;
 	}
 }
+
+enum pkcs11_rc pkcs2tee_param_ecdh(struct pkcs11_attribute_head *proc_params,
+				   void **pub_data, size_t *pub_size)
+{
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+	struct serialargs args = { };
+	uint32_t temp = 0;
+
+	serialargs_init(&args, proc_params->data, proc_params->size);
+
+	/* Skip KDF */
+	rc = serialargs_get(&args, &temp, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	/* Shared data size, shall be 0 */
+	rc = serialargs_get(&args, &temp, sizeof(uint32_t));
+	if (rc || temp)
+		return rc;
+
+	/* Public data size and content */
+	rc = serialargs_get(&args, &temp, sizeof(uint32_t));
+	if (rc || !temp)
+		return rc;
+
+	*pub_size = temp;
+
+	rc = serialargs_get(&args, &temp, sizeof(uint8_t));
```

Comment:
`temp` is a `uint32_t`, you should probably use some temporary `uint8_t` variable instead.